docs/source/ddp_comm_hooks.rst

```diff
@@ -0,0 +1,64 @@
+DDP Communication Hooks
+=======================
+
+DDP communication hook is a generic interface to control how to communicate
+gradients across workers.
+A few built-in communication hooks are provided,
+and users can easily apply any of these hooks to optimize communication.
+Besides, the hook interface can also support user-defined communication
+strategies for more advanced use cases.
+
+.. warning ::
+    DDP communication hooks can only support single process single device mode
+    on NCCL backend.
+
+How to Use A Communication Hook?
+--------------------------------
+
+To use a communication hook, the user just needs to let the DDP model register
+the hook before the training loop.
+
+.. automethod:: torch.nn.parallel.DistributedDataParallel.register_comm_hook
```

Comment:
Can you also add a code example for how a user might define an allreduce hook, and then call it with a state object?

Reply:
I believe the docstrings for `register_comm_hook` do have a couple of examples.

Reply:
Yes, already `allreduce_hook` already has an example in the docstrings.